### PR TITLE
Feature/bp pythonserver

### PR DESCRIPTION
**Pull request by pcrespov**
Added in the boilerplate two different python servers as back-end. In both cases, qx-client is served and communicate with socket.io.  

- Thrift interface still missing but will add it as soon as SyA pass me the python API for that.
- All examples are dockerized and pass pylint
- Did some tuning to travis and pylint setup

### CODE REVIEW

**Comment by sanderegg** [comment]
Hey Pedro,

Please find the Python Thrift library and the interfaces to XRpc Application, Modeler and Studies in your sandbox.
Also there a jupyter notebook with an example how to use them.

But on a side note, I saw there might be a special flag for using Thrift with Tornado using websockets... I-ll investigate tomorrow.

**Comment by pcrespov** [comment]
Thx Sylvain

File: tools/ci-checks/test_demos_file_structure.py [hunk 1]
```diff
@@ -77,7 +79,7 @@ def test_file_structure():
     for sub_folder in os.listdir('demos'):
         folder = os.path.join(root_folder, sub_folder)
         if os.path.isdir(folder):
-            logging.info('Testing folder: ' + folder)
+            logging.info('Testing folder: %s', folder)
```
Comment (sbenkler) [inline tools/ci-checks/test_demos_file_structure.py]: Is this pep8 or just about style (concat string vs. using format capabilities)

Comment (pcrespov) [reply tools/ci-checks/test_demos_file_structure.py]: This is a well know pitfall ... logger formatter will catch errors while formatting concatenation witll raise them ... see example discussion [here](https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html)

**Pull request merged**